codec-http/src/main/java/io/netty/handler/codec/http/HttpContentDecoder.java

```diff
@@ -85,7 +85,15 @@ protected void decode(ChannelHandlerContext ctx, HttpObject msg, List<Object> ou
                 if (contentEncoding != null) {
                     contentEncoding = contentEncoding.trim();
                 } else {
-                    contentEncoding = IDENTITY;
+                    String transferEncoding = headers.get(HttpHeaderNames.TRANSFER_ENCODING);
+                    if (transferEncoding != null) {
+                        contentEncoding = transferEncoding.trim();
+                        if (transferEncoding.contains(",")) {
+                            contentEncoding = transferEncoding.split(",")[0].trim();
+                        }
```

Comment:
This can be changed to the following code to reduce the `trim()` operations and also reduce allocations:

```java
int idx = transferEncoding.indexOf(",");
if (idx != -1) {
    contentEncoding = transferEncoding.subString(0, idx).trim();
} else {
    contentEncoding = transferEncoding.trim();
}
```

Reply:
Thanks for the advice, Done